x/ecocredit/server/core/query_projects.go

```diff
@@ -0,0 +1,44 @@
+package core
+
+import (
+	"context"
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	ecocreditv1beta1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1beta1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+)
+
+// Projects queries all projects from a given credit class.
+func (k Keeper) Projects(ctx context.Context, request *v1beta1.QueryProjectsRequest) (*v1beta1.QueryProjectsResponse, error) {
+	pg, err := GogoPageReqToPulsarPageReq(request.Pagination)
+	if err != nil {
+		return nil, err
+	}
+	cInfo, err := k.stateStore.ClassInfoStore().GetByName(ctx, request.ClassId)
+	if err != nil {
+		return nil, err
+	}
+	it, err := k.stateStore.ProjectInfoStore().List(ctx, ecocreditv1beta1.ProjectInfoClassIdNameIndexKey{}.WithClassId(cInfo.Id), ormlist.Paginate(pg))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
We only need the id from class info, which we already have in the request. Do we need to get class info from state or can we just convert the id in the request?

Reply:
nvm